### ISSUE TITLE
Batch Normalization doesn't support leaky relu post op?

### ISSUE THREAD

**Issue by congkun-ck** [oneapi-src/oneDNN#1143]
# Summary
Trying to add a leaky relu (slope=0.01) post op to BatchNorm fails with unimplemented.

# Version
dnnl_verbose,info,oneDNN v2.3.0 (commit 3420bb12ffea5f17e90f73bc2232ed09c79dad92)

# Environment
Windows 10 x64
dnnl_verbose,info,cpu,runtime:OpenMP
dnnl_verbose,info,cpu,isa:Intel AVX2
dnnl_verbose,info,gpu,runtime:none

# Steps to reproduce
The following code fails with unimplemented error. 
It tries to create a BatchNorm primitive desc with a post op of leaky relu (eltwise relu-algo alpha=0.01f). 
For normal relu (alpha=0.0f) it works fine.

```
dnnl::set_max_cpu_isa(dnnl::cpu_isa::avx2);
dnnl::engine engine(dnnl::engine::kind::cpu, 0);
dnnl::stream stream(engine, dnnl::stream::flags::default_flags);

dnnl::memory::desc test_desc{ { { 1, 1, 2, 257 } }, dnnl::memory::data_type::f32, dnnl::memory::format_tag::nchw };
auto bnr_desc = dnnl::batch_normalization_forward::desc(dnnl::prop_kind::forward_inference, test_desc, 1e-5f, dnnl::normalization_flags::use_scale_shift | dnnl::normalization_flags::use_global_stats);
dnnl::primitive_attr bnr_attr;
dnnl::post_ops bnr_post_ops;
bnr_post_ops.append_eltwise(1.0f, dnnl::algorithm::eltwise_relu, 0.01f, 0.0f); //<-- alpha=0.01 fails
//bnr_post_ops.append_eltwise(1.0f, dnnl::algorithm::eltwise_relu, 0.0f, 0.0f); //<-- alpha=0.0f succeeds
bnr_attr.set_post_ops(bnr_post_ops);
auto bnr_prim_desc = dnnl::batch_normalization_forward::primitive_desc(bnr_desc, bnr_attr, engine); //<-- fails with unimplemented error
```

# Observed behavior
It fails with unimplemented error for leaky relu post op.

# Expected behavior
It should work.

**Comment by dzarukin** [comment]
Hi @congkun-ck , thank you for the report. This is an expected unimplemented case. The rationale for such behavior is the following:
1) In training mode using leaky relu doesn't allow to fuse safely since an intermediate result is required which wouldn't be produced in case of post-ops. That's why only zero slope is supported since this fusion can be gracefully processed on backward. For cases where intermediate result is required, a dedicated eltwise primitive should be created to handle such case.
2) For inference (in your example) in most cases mean and variance are pre-computed and should be passed as input using a special flag `dnnl_use_global_stats`. For such scenario, binary primitive with binary post-ops on top may be used instead, which supports fusion on inference for leaky relu and any other type of post-operations.
3) Rare scenario when on inference one needs to actually compute mean and variance. If this is your case, could you provide more details where and why it happens? As a quick work around I'd recommend to follow the recipe from (1) until the situation is clear.
Thank you.

**Comment by congkun-ck** [comment]
Thanks @dzarukin . That makes sense.
